docs/dev/Kibana-Reporting-Design-Proposal.md

```diff
@@ -20,43 +21,111 @@ Kibana Reports for Open Distro allows ‘Report Owner’ (engineers, including b
+    1. As a user, if I select `custom cron`, I should be able to configure a cron-based schedule.
+    1. As a user, if I select `recurring` or `custom cron`, I should be able to configure an end date.
+1. As a user, when I select `Alert`, I should be able to select the alert a report gets triggered from.
+
+##### Delivery settings
+1. As a user, I should be able to deliver a report to Kibana and/or Email recepients.
+1. As a user, when I set Kibana as a delivery channel, I should be able to select which Kibana users the report gets delivered to.
+1. As a user, when I set email as a delivery channel, I should be able to set recepients, email subject, and email body.
+1. As a user, I should be able to insert a file URL refecence to the email body.
+1. As a user, I should be able to insert the report source URL reference to the email body.
+1. As a user, I should be able to attach the report file to an email.
+1. As a user, I should be able to set the file format of report attachment (pdf, png, csv, excel, contextually).
+
+#### Report creation
+1. As a user, when I create a report definition, the definition settings are saved and show up as a list item in the ‘Report definitions’ list.
+1. As a user, when a report definition has been triggered via schedule or alert, I should be able to see an instance of the report in my "Reports" list, and any users I send the report to, will also see a report instance in their repective "Reports" list page.
+1. As a user, when a report is created from a triggered definition condition, I should receive a notification on all configured delivery channels (Email, chime, slack, etc)
+1. As a user, when a report is created from a triggered definition condition, I should receive a kibana toast alert letting me know that the report is available.
+1. As a user, when a report fails due to an error, I should receive a kibana toast alert letting me know that the report failed.
+1. As a user, when a report fails due to an error, I should see the report with a failed status on the `Reports` list.
+
+#### View report definition details
+1. As a user, I should be able to view all my saved definition settings.
+1. As a user, I should be able to edit, duplicate or delete a report definition.
+    1. As a user, when I delete a report definition, any schedule or alert based triggers will automatically stop.
+
+#### Edit report definition
+1. As a user, I should be able to edit the report settings, triggers and delivery settings.
+1. As a user, I should be able to save my updated report definition.
+
+#### View report details
+1. As a user, I should be able to see both created date, and last updated date.
+1. As a user, I should be able to share an existing report.
+1. As a user, I should be able to archive an existing report. (p2)
+1. As a user, I should be able to downlaod available report files (PDF, PNG, CSV, Excel, etc)
+1. As a user, I should be able to copy the permalink to a report source.
```

Comment:
Do we need to mention this here? Because permalink is a feature provided by native Kibana

Reply:
Yes, from the report details page, we have a "Report source" field that has a link to the source dashboard. We're providing a "copy to clipboard" action next to the report source value field to make it easier to click through /copy to the source.